server/valgrind-out.txt

```diff
@@ -1,216 +1,257 @@
+==58070== 
+==58070== 
+==58070== 1 errors in context 2 of 5:
+==58070== Syscall param rt_sigaction(act->sa_mask) points to uninitialised byte(s)
+==58070==    at 0x48BB5F3: __libc_sigaction (libc_sigaction.c:58)
+==58070==    by 0x10A7D9: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070==  Address 0x1ffefffb48 is on thread 1's stack
+==58070==  in frame #0, created by __libc_sigaction (libc_sigaction.c:43)
+==58070==  Uninitialised value was created by a stack allocation
+==58070==    at 0x10A6F1: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070== 
+==58070== 
+==58070== 1 errors in context 3 of 5:
+==58070== Syscall param rt_sigaction(act->sa_flags) points to uninitialised byte(s)
+==58070==    at 0x48BB5F3: __libc_sigaction (libc_sigaction.c:58)
+==58070==    by 0x10A7A2: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070==  Address 0x1ffefffb38 is on thread 1's stack
+==58070==  in frame #0, created by __libc_sigaction (libc_sigaction.c:43)
+==58070==  Uninitialised value was created by a stack allocation
+==58070==    at 0x10A6E9: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070== 
+==58070== 
+==58070== 1 errors in context 4 of 5:
+==58070== Syscall param rt_sigaction(act->sa_mask) points to uninitialised byte(s)
+==58070==    at 0x48BB5F3: __libc_sigaction (libc_sigaction.c:58)
+==58070==    by 0x10A7A2: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070==  Address 0x1ffefffb48 is on thread 1's stack
+==58070==  in frame #0, created by __libc_sigaction (libc_sigaction.c:43)
+==58070==  Uninitialised value was created by a stack allocation
+==58070==    at 0x10A6F1: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070== 
+==58070== 
+==58070== 17 errors in context 5 of 5:
+==58070== Thread 2:
+==58070== Conditional jump or move depends on uninitialised value(s)
+==58070==    at 0x484ED28: strlen (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==58070==    by 0x48EFD30: __vfprintf_internal (vfprintf-internal.c:1517)
+==58070==    by 0x4996FB9: __vsyslog_internal (syslog.c:229)
+==58070==    by 0x4997499: syslog (syslog.c:109)
+==58070==    by 0x10B197: joiner_handler (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
+==58070==    by 0x490DAC2: start_thread (pthread_create.c:442)
+==58070==    by 0x499EA03: clone (clone.S:100)
+==58070==  Uninitialised value was created by a heap allocation
+==58070==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==58070==    by 0x10AB9E: main (in /home/mueller4110/Documents/embedded_linux/assignment-2-almu6129/server/aesdsocket)
```

Comment:
Most if not all of the issues identified here are likely real.  If you rebuild with `-g` it will put line numbers on each one of these for you to help track them down.